diff --git a/ProyectoFinal/AppCoder/forms.py b/ProyectoFinal/AppCoder/forms.py
index 7afdd77..afd1326 100644
--- a/ProyectoFinal/AppCoder/forms.py
+++ b/ProyectoFinal/AppCoder/forms.py
@@ -4,9 +4,9 @@ class UsuarioForm (forms.Form):
 
     nombre =  forms.CharField (max_length=40)
     apellido = forms.CharField(max_length=40)
-    fechaNacimiento = forms.DateField()
-    contrasena = forms.CharField(widget=forms.PasswordInput,max_length=40)
-
+    fechaNacimiento = forms.DateField(required=False)
+    email = forms.EmailField(required=True)
+    contrasena = forms.CharField(widget=forms.PasswordInput,max_length=40)    
 
 class PeliculaForm(forms.Form):
 
diff --git a/ProyectoFinal/AppCoder/templates/AppCoder/registrook.html b/ProyectoFinal/AppCoder/templates/AppCoder/registrook.html
index 677779b..dbcd620 100644
--- a/ProyectoFinal/AppCoder/templates/AppCoder/registrook.html
+++ b/ProyectoFinal/AppCoder/templates/AppCoder/registrook.html
@@ -1,18 +1,18 @@
-{% extends "AppCoder/madre.html" %}
+{% extends "AppCoder/madre.html" %} {% load static %} {% block contenido %}
 
-{% load static %}
-
-
-{% block contenido %}
-
-<br>
-<br>
+<br />
+<br />
 
+{% if not errors %}
 <div class="text-center">
-    <h2 class="fw-bolder">Usted se ha registado en el mejor blog del mundo</h2>
-    <p class="lead fw-normal text-muted mb-5">Aplausos!</p>
+  <h2 class="fw-bolder">Usted se ha registado en el mejor blog del mundo</h2>
+  <p class="lead fw-normal text-muted mb-5">Aplausos!</p>
 </div>
-
-
-
-{% endblock %}
\ No newline at end of file
+{% else %}
+<div class="text-center">
+  <h2 class="fw-bolder">Aviso, datos incorrectos:</h2>
+  <p class="lead fw-normal text-muted mb-5">
+    {% for error in errors %} {{ error }} {% endfor %}
+  </p>
+</div>
+{% endif %} {% endblock %}
diff --git a/ProyectoFinal/AppCoder/templates/AppCoder/userFormulario.html b/ProyectoFinal/AppCoder/templates/AppCoder/userFormulario.html
index c264c29..8477b4c 100644
--- a/ProyectoFinal/AppCoder/templates/AppCoder/userFormulario.html
+++ b/ProyectoFinal/AppCoder/templates/AppCoder/userFormulario.html
@@ -1,49 +1,30 @@
+{% extends "AppCoder/madre.html" %} {% load static %} {% block contenido %}
 
-{% extends "AppCoder/madre.html" %}
+<br />
+<br />
+<br />
 
-{% load static %}
+<div class="text-center">
+  <h2 class="fw-bolder">Registrate</h2>
+  <p class="lead fw-normal text-muted mb-5">
+    Registrate para acceder a nuevas posibilidades.
+  </p>
 
-
-        {% block contenido %}
-
-        <br>
-        <br>
-        <br>
-
-        <div class="text-center">
-
-        <h2 class="fw-bolder">Registrate</h2>
-        <p class="lead fw-normal text-muted mb-5">Registrate para acceder a nuevas posibilidades.</p>
-
-        <form action="/AppCoder/userFormulario/" method="POST">{% csrf_token %}
-            <!--
+  <form action="{% url 'userFormulario' %}" method="POST">
+    {% csrf_token %}
+    <!--
             <p>Nombre: <input type="text" name="nombre")></p>
             <p>Edad: <input type="number" name="edad")></p>
             <p>Direccion:  <input type="text" name="direccion")></p>
             <p>FechaNacimiento: <input type="datetime" name="fechaNacimiento")></p>
             -->
 
-            <table>
-
-                {{miFormulario.as_table}}
-
-
-            </table>
-           
-
-            <input type="submit" value="Registrar">
-
-
-
-
-
-        </form>
-
-        </div>
-
-        {% endblock %}
-
-
+    <table>
+      {{ miFormulario.as_table }}
+    </table>
 
+    <input type="submit" value="Registrar" />
+  </form>
+</div>
 
-        
\ No newline at end of file
+{% endblock %}
diff --git a/ProyectoFinal/AppCoder/urls.py b/ProyectoFinal/AppCoder/urls.py
index 7da40f2..0751752 100644
--- a/ProyectoFinal/AppCoder/urls.py
+++ b/ProyectoFinal/AppCoder/urls.py
@@ -4,7 +4,7 @@ from django.contrib.auth.views import LogoutView
 
 urlpatterns = [
 
-    path("inicio/", views.inicio, name="Inicio"),
+    path("", views.inicio, name="Inicio"),
     path("sobrenosotrxs/", views.sobreNosotrxs, name="Sobre nosotrxs"),
     path("post1/", views.post1, name="Post1"),
     path("contacto/", views.contacto, name="Contacto"),
@@ -17,5 +17,4 @@ urlpatterns = [
     path('logout', LogoutView.as_view(template_name='AppCoder/logout.html'), name="Logout"),
     path('editarPerfil', views.editarPerfil, name="EditarPerfil"),
     path('agregarAvatar', views.agregarAvatar, name="AgregarAvatar"),
-
 ]
\ No newline at end of file
diff --git a/ProyectoFinal/AppCoder/views.py b/ProyectoFinal/AppCoder/views.py
index f1776ce..8d60f8a 100644
--- a/ProyectoFinal/AppCoder/views.py
+++ b/ProyectoFinal/AppCoder/views.py
@@ -38,15 +38,22 @@ def contacto(request):
     return render(request, "AppCoder/contacto.html")
 
 def userFormulario(request):
-
+    
+    context = {}
     if request.method == "POST":
 
-        miFormulario = UsuarioForm (request.POST)
-        if miFormulario.is_valid(): 
-            userInsta = usuario (nombre=request.POST["nombre"], apellido=request.POST["apellido"],fechaNacimiento = request.POST["fechaNacimiento"], contrasena=request.POST["contrasena"]  )  
+        miFormulario = UsuarioForm(request.POST)
+        if miFormulario.is_valid():
+            userInsta = usuario(
+                nombre=miFormulario.cleaned_data.get("nombre"),
+                apellido=miFormulario.cleaned_data.get("apellido"),
+                fechaNacimiento = miFormulario.cleaned_data.get("fechaNacimiento"),
+                contrasena=miFormulario.cleaned_data.get("contrasena")
+            )
             userInsta.save() #Guarda en la base de datos
-
-        return render(request, "AppCoder/registrook.html")
+        else:
+            context['errors'] = miFormulario.errors
+        return render(request, "AppCoder/registrook.html", context)
     
     else:
         miFormulario = UsuarioForm()
@@ -145,12 +152,12 @@ def register(request):
             
             if form.is_valid():
 
-                  username = form.cleaned_data['username']
-                  
+                username = form.cleaned_data['username']
                   
-                  form.save()
+                form.save()
                   
-                  return render(request,"AppCoder/inicio.html" ,  {"mensaje":f"{username} Creado :)"})
+                return render(request,"AppCoder/inicio.html" ,  {"mensaje":f"{username} Creado :)"})
+
 
 
       else:
diff --git a/ProyectoFinal/ProyectoFinal/urls.py b/ProyectoFinal/ProyectoFinal/urls.py
index 3f280c4..1e983f7 100644
--- a/ProyectoFinal/ProyectoFinal/urls.py
+++ b/ProyectoFinal/ProyectoFinal/urls.py
@@ -16,8 +16,8 @@ Including another URLconf
 from django.contrib import admin
 from django.urls import path, include
 
+
 urlpatterns = [
     path('admin/', admin.site.urls),
-    path("AppCoder/", include("AppCoder.urls"))
-
+    path("", include("AppCoder.urls")),
 ]
